File: DynamoDB-SDK-Examples/node.js/WorkingWithItems/get-item.js
// This is an example of a simple GetCommand with the higher level DocumentClient for Amazon DynamoDB

const { DynamoDBClient } = require("@aws-sdk/client-dynamodb");
const { DynamoDBDocumentClient, GetCommand } = require("@aws-sdk/lib-dynamodb");

async function getItems() {
  const client = new DynamoDBClient({ region: "us-west-2" });
  const ddbDocClient = DynamoDBDocumentClient.from(client);
  try {
    return await ddbDocClient.send(
        new GetCommand({
          TableName: "RetailDatabase",
          Key: {
            pk: "jim.bob@somewhere.com", // Partition Key
            sk: "metadata"          // Sort Key
          },
          // For this use case, the data does not changed often so why not get the
          // reads at half the cost? Your use case might be different and need true.
          ConsistentRead: false,
        })
    );
  } catch (err) {
    console.error(err);
  }
}

getItems()
    .then((data) =>
        console.log("GetCommand succeeded:", JSON.stringify(data.Item, null, 2)))
    .catch((error) => console.error(JSON.stringify(error, null, 2)));

File: examples/SDK/kotlin/data_plane/WorkingWithItems/src/main/kotlin/GetItem.kt
import aws.sdk.kotlin.services.dynamodb.DynamoDbClient
import aws.sdk.kotlin.services.dynamodb.getItem
import aws.sdk.kotlin.services.dynamodb.model.AttributeValue

suspend fun getItem(
    tableName: String,
    key: Map<String, AttributeValue>,
) {
    DynamoDbClient { region = "us-west-2" }.use { ddb ->
        val response =
            ddb.getItem {
                this.tableName = tableName
                this.key = key
            }

        response.item?.let {
            println("Item: $it")
        } ?: println("No item found with the given key.")
    }
}

suspend fun main() {
    val tableName = "YourTableName"
    val key = mapOf("id" to AttributeValue.S("1234"))

    getItem(tableName, key)
}
